<launch>
  <!-- 启动仿真世界（如果你需要） -->
  <include file="$(find me5413_world)/launch/world.launch" />
  
  <!-- Connect the robot to a keyboard teleop controller -->
  <node name="teleop_twist_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" output="screen" respawn="true"/>

  <!-- Run the map server -->
  <arg name="map_file" default="$(find jackal_navigation)/maps/utm30.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
  <!--node name="ekf_localization" pkg="robot_localization" type="ekf_localization_node"/-->
  
  <!-- Launch the AMCL Localizer -->
  <include file="$(find jackal_navigation)/launch/include/amcl_040202.launch" />

  <!-- Launch Move Base -->
  <include file="$(find jackal_navigation)/launch/include/move_base.launch" />

  <!-- Launch Rviz with our settings -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find me5413_world)/rviz/navigation.rviz" output="log" respawn="true"/>

  <node ns="me5413_world" pkg="me5413_world" type="goal_publisher_node" name="goal_publisher_node" output="screen" />

  <!-- 自动发布目标点脚本 -->
  <!-- node name="auto_goal_publisher" pkg="jackal_navigation" type="bridge_navigation.py" output="screen" / -->
  
  <!-- node name="cube_merged_node" pkg="jackal_navigation" type="cube.py" output="screen" / -->
  
  
  <!-- 过桥 -->
  <node name="bridge_detector" pkg="jackal_navigation" type="cube_bridge_detector.py" output="screen" />
  <node name="bridge_patrol_and_responder" pkg="jackal_navigation" type="patrol_and_cross_bridge.py" output="screen" />
  
  <!-- 过桥之后 -->
  <node name="cube_merged_node" pkg="jackal_navigation" type="cube_after_bridge.py" output="screen" />
  <node pkg="jackal_navigation" type="end_navigation.py" name="end_navigation" output="screen">
    <param name="target_box_number" value="3" />
  </node>
  
  
</launch>
